### PR TITLE
Fixes for launching with Eclipse

### PR DESCRIPTION
**Pull request by briandealwis**
With these fixes I can:
  - create a test app
  - adjust changes in the project property page
  - Debug As > GWT Super Dev Mode
  - Double-click in Development Mode view to launch in browser

### CODE REVIEW

File: plugins/com.gwtplugins.gwt.eclipse.oophm/src/com/google/gwt/eclipse/oophm/views/hierarchical/LaunchConfigurationContent.java [hunk 1]
```diff
@@ -269,7 +269,7 @@ private void createViewer(Label launchUrlsCaptionLabel) {
     viewer.addDoubleClickListener(new IDoubleClickListener() {
       @Override
       public void doubleClick(DoubleClickEvent event) {
-
+        browserMenuPopulator.openDefaultBrowser(getProject(launchConfiguration), getSelectedUrl());
```
Comment (branflake2267) [inline plugins/com.gwtplugins.gwt.eclipse.oophm/src/com/google/gwt/eclipse/oophm/views/hierarchical/LaunchConfigurationContent.java]: I was wondering why that was.

File: plugins/com.gwtplugins.gwt.eclipse.core/src/com/google/gwt/eclipse/core/preferences/ui/GwtPreferencePage.java [hunk 2]
```diff
@@ -42,12 +42,12 @@
 public class GwtPreferencePage extends PreferencePage implements
     IWorkbenchPreferencePage {
 
-  public static final String ID = GwtPreferencePage.class.getName();
+  public static final String ID = "com.gwtplugins.gwt.eclipse.core.preferences.ui.GwtPreferencePage";
```
Comment (branflake2267) [inline plugins/com.gwtplugins.gwt.eclipse.core/src/com/google/gwt/eclipse/core/preferences/ui/GwtPreferencePage.java]: I noticed this too, thanks!

**Comment by branflake2267** [comment]
Thanks for the fixes and enhancements!

File: plugins/com.gwtplugins.gwt.eclipse.core/plugin.xml [hunk 3]
```diff
@@ -706,6 +706,11 @@
             class="com.google.gwt.eclipse.core.properties.ui.GWTProjectPropertyPage"
             id="com.gwtplugins.gwt.eclipse.core.properties.ui.gwtProjectPropertyPage"
             name="General Settings">
+         <enabledWhen>
```
Comment (branflake2267) [inline plugins/com.gwtplugins.gwt.eclipse.core/plugin.xml]: This was being greedy, turning it off with a java project. I found another way to do it. 
https://github.com/gwt-plugins/gwt-eclipse-plugin/pull/223/files#diff-bf86cc2cb0c38b43ba602df9a5e590acR710

```
<enabledWhen>
   <adapt type="org.eclipse.core.resources.IProject">
      <test property="org.eclipse.core.resources.projectNature"
            value="org.eclipse.jdt.core.javanature"/>
   </adapt>         
 </enabledWhen>

Comment (briandealwis) [inline plugins/com.gwtplugins.gwt.eclipse.core/plugin.xml]: Hmm… `GWTProjectPropertyPage` assumes [an `IJavaProject`](https://github.com/gwt-plugins/gwt-eclipse-plugin/blob/master/plugins/com.gwtplugins.gwt.eclipse.core/src/com/google/gwt/eclipse/core/properties/ui/GWTProjectPropertyPage.java#L247).

Ah the page is a subpage of `com.gwtplugins.gdt.eclipse.core.properties.coreProjectPropertyPage` (defined in `.gdt.eclipse.core`) which is conditioned on the `com.gwtplugins.gdt.eclipse.core.isGpeSelected` expression, [defined as](https://github.com/gwt-plugins/gwt-eclipse-plugin/blob/master/plugins/com.gwtplugins.gdt.eclipse.core/plugin.xml#L12):
```
        <definition id="com.gwtplugins.gdt.eclipse.core.isGpeSelected">
            <and>
               <adapt type="org.eclipse.core.resources.IProject">
                   <test property="org.eclipse.core.resources.projectNature"
                         value="org.eclipse.jdt.core.javanature">
                   </test>
               </adapt>
            </and>
        </definition>
```
So it doesn't appear on non-Java projects anyways.

Comment (branflake2267) [inline plugins/com.gwtplugins.gwt.eclipse.core/plugin.xml]: Do you think if com.gwtplugins.gdt.eclipse.core.isGpeSelected was changed to enabledWhen IJavaProject it should work like you expected?

**Pull request merged**